src/classes/OPP_CreateTribute_BATCH.cls

```diff
@@ -0,0 +1,114 @@
+/*
+    Copyright (c) 2018, Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2018
+* @group Tribute
+* @description Batch Class for creating Tribute from Opportunity.
+*/
+public class OPP_CreateTribute_BATCH implements Database.Batchable<sObject> {
+
+    /*******************************************************************************************************
+    * @description Opportunity Query used to find Opportunities to Convert
+    */ 
+    private static final String OPP_QUERY = 'SELECT Id, Tribute_Type__c, Honoree_Contact__c, Honoree_Name__c ' +
```

Comment:
Ok, makes sense.   I will move the null check to the batch, and skip ones that do not match.  I am leaning toward requerying for the data too, for each batch set-- if this is running over millions of records it is possible it was updated between when the query ran and the batch is processing.

Reply:
@force2b - If I do that, it is constraint in size to the set of records I am processing.  I can do an if in a loop, and build a new collection (after retrieving them, or I can just filter out the ones that are null (since it is no longer unbounded).  What is your preference?